### PR TITLE
Add kci_rootfs upload option

### PR DESCRIPTION
**Pull request by Lakshmipathi**
Upload rootfs images to kernelci backend.
Usage: ./kci_rootfs upload --rootfs-dir=/path/to/rootfsdir
       --upload-path=path/to/uploaddir --api=ip-address --token <token>

Signed-off-by: Lakshmipathi Ganapathi <lakshmipathi.ganapathi@collabora.com>

### CODE REVIEW

File: kernelci/rootfs.py [hunk 1]
```diff
@@ -50,3 +52,37 @@ def build(name, config, data_path):
     else:
         print("rootfs_type:{} not supported".format(config.rootfs_type))
         return False
+
+
+def _upload_files(api, token, path, input_files):
```
Comment (mgalka) [inline kernelci/rootfs.py]: There is a very similar function defined in `kernelci/build.py`. We should avoid this kind of a duplication. I think it's a good idea to create a `kernelci/storage.py` module and put the functions responsible for calling the /upload API and import it in both `rootfs.py` and `build.py`.

Comment (Lakshmipathi) [reply kernelci/rootfs.py]: okay added required changes to `rootfs.py` and `build.py`

File: kernelci/rootfs.py [hunk 2]
```diff
@@ -16,7 +16,10 @@
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
 from kernelci import shell_cmd
+from kernelci.storage import upload_files
 import os
+import requests
+from urllib.parse import urljoin
```
Comment (mgalka) [inline kernelci/rootfs.py]: These imports seem to be unused

**Comment by mgalka** [comment]
Tested locally, works fine.

File: kernelci/storage.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        'file{}'.format(i): (name, fobj)
+        for i, (name, fobj) in enumerate(input_files.items())
+    }
+    url = urljoin(api, 'upload')
+    resp = requests.post(url, headers=headers, data=data, files=files)
+    resp.raise_for_status()
+    return True
+
+
+def upload_files(api, token, upload_path, input_files):
```
Comment (mgalka) [inline kernelci/storage.py]: This function does nothing apart from calling the non-public API (_upload_files()). We can get rid of this function and rename `_upload_files` -> `upload_files`.
The docstring should be also applied to the renamed function.

File: kernelci/storage.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    """Upload rootfs to KernelCI backend.
+
+    *api* is the URL of the KernelCI backend API
+    *token* is the backend API token to use
+    *upload_path* is the target on KernelCI backend
+    *input_files* dictionary of input files
+    """
+    _upload_files(api, token, upload_path, input_files)
+    return True
```
Comment (mgalka) [inline kernelci/storage.py]: This function always returns `True` and doesn't really care about the result.

Comment (Lakshmipathi) [reply kernelci/storage.py]: `raise_for_status()` call should handle the negative case and should report any exception.

Comment (gctucker) [reply kernelci/storage.py]: Then, there's no need to return anything.  That's what the original `_upload_files()` did.

**Comment by gctucker** [comment]
OK starting a Buster job on staging today as a one last check before merging.

**Comment by gctucker** [comment]
Sorry, forgot this isn't integrated with Jenkins yet on the master branch.  So yes it seems all ready to go then.

**Pull request merged**